compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h

```diff
@@ -32,42 +33,41 @@ void addPadBasedPassPipeline(OpPassManager &passManager,
 
 /// Create a pass to do some rewrites that help bridging the path to AIR/AIE
 /// lowering.
-std::unique_ptr<OperationPass<>> createAMDAIEBridgeToAIRPass();
+std::unique_ptr<Pass> createAMDAIEBridgeToAIRPass();
 
 /// Pass to bufferizes the targeted operation and materializes the result in a
 /// new allocation.
-/// Currently using paddingLevel = 0 to indicate the target operation is
-/// packing.
-/// TODO: Find a better way to do the pass without using paddingLevel.
+std::unique_ptr<Pass> createAMDAIEBufferizeToAllocationPass();
 std::unique_ptr<OperationPass<func::FuncOp>>
-createAMDAIEBufferizeToAllocationPass(int64_t memorySpace = 1,
-                                      int64_t paddingLevel = 0);
+createAMDAIEBufferizeToAllocationPass(
+    const AMDAIEBufferizeToAllocationOptions &options);
 
 /// Create pass to invoke several cleanup and canonicalization patterns.
-std::unique_ptr<OperationPass<func::FuncOp>> createAMDAIECleanupPass();
+std::unique_ptr<Pass> createAMDAIECleanupPass();
 
 /// Create pass calling the dynamic pipeline for AMDAIE.
-std::unique_ptr<OperationPass<IREE::HAL::ExecutableVariantOp>>
-createAMDAIELowerExecutableTargetPass();
+std::unique_ptr<Pass> createAMDAIELowerExecutableTargetPass();
 
 /// Create a pass to lower workgroup count region of entry point operations.
-std::unique_ptr<OperationPass<IREE::HAL::ExecutableVariantOp>>
-createAMDAIELowerWorkgroupCountPass();
+std::unique_ptr<Pass> createAMDAIELowerWorkgroupCountPass();
 
 /// Create a pass to pack and transpose the linalg op.
+std::unique_ptr<Pass> createAMDAIEPackAndTransposePass();
 std::unique_ptr<OperationPass<func::FuncOp>> createAMDAIEPackAndTransposePass(
-    int64_t packLevel = 1);
+    const AMDAIEPackAndTransposeOptions &options);
```

Comment:
can default values to options not be provided with these refactor?

Reply:
The `.td` will take care of the default values. The beauty of using the newer base class is that even if we add/remove options, we don't have to change the function declarations and invocations - just the newer/old option parameter needs to be added/removed right before invoking the pass.